### PR TITLE
[VL] Update Velox build script

### PR DESCRIPTION
**Pull request by Mionsz**
## What changes were proposed in this pull request?

Removed `-mno-avx512f` flag from build process.
Update Ubuntu build script human-readable.
Remove `sed` from wherever it is possible.
Make the build process more configurable.
Enable all options to be read from environment variables.
Move all commands to after the functions definitions. 
Make setting `nproc` configurable - else it will often fail in cloud builds. 
Some other improvements.

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

### CODE REVIEW

**Comment by zhejiangxiaomai** [comment]
I think "-mno-avx512f" is necessary for velox build. In our machine, we will get error. 

``` ../.././velox/common/base/SimdUtil-inl.h: In instantiation of ‘int32_t facebook::velox::simd::indicesOfSetBits(const uint64_t*, int32_t, int32_t, int32_t*, const A&) [with A = xsimd::avx512bw; int32_t = int; uint64_t = long unsigned int]’:
../../velox/common/base/tests/SimdUtilTest.cpp:65:70:   required from here
../.././velox/common/base/SimdUtil-inl.h:222:39: error: static assertion failed
  222 |             static_assert(Batch::size == 4);
      |                                  ~~~~~^~~~
```

File: ep/build-velox/src/build_velox.sh [hunk 1]
```diff
@@ -99,53 +145,32 @@ function process_setup_centos8 {
...
-  exit 1
-fi
-
 function process_script {
     sed -i 's/^  ninja -C "${BINARY_DIR}" install/  sudo ninja -C "${BINARY_DIR}" install/g' scripts/setup-helper-functions.sh
-    sed -i 's/-mavx2 -mfma -mavx -mf16c -mlzcnt -std=c++17/-march=native -std=c++17 -mno-avx512f/g' scripts/setup-helper-functions.sh
-    if [[ "$LINUX_DISTRIBUTION" == "ubuntu" || "$LINUX_DISTRIBUTION" == "debian" ]]; then
-      process_setup_ubuntu
-    else # Assume CentOS
+    sed -i "s/-mavx2 -mfma -mavx -mf16c -mlzcnt -std=c++17/-march=${BUILD_MARCH} -std=c++17/g" scripts/setup-helper-functions.sh
```
Comment (zhejiangxiaomai) [inline ep/build-velox/src/build_velox.sh:150]: we should not remove -mno-avx512f.

**Comment by Mionsz** [comment]
@zhejiangxiaomai  Done.

**Comment by zhejiangxiaomai** [comment]
--But we should have a way to enable avx512 gcc flag. The worst way is to add --target_arch=SKL/ICX/CLX/SPR, etc. to build commandline.

Ok, I see.

**Comment by zhejiangxiaomai** [comment]
> @zhejiangxiaomai Done.

The CI still have some error. I think you can use ./dev/buildbundle-veloxbe.sh to debug error. Because this script will install gluten+velox.

**Pull request closed without merging**